eth/executionclient/execution_client.go

```diff
@@ -30,35 +31,44 @@ var (
 type ExecutionClient struct {
 	// mandatory
 	nodeAddr        string
+	blocksChan      chan uint64
 	contractAddress ethcommon.Address
 
 	// optional
-	logger                      *zap.Logger
-	metrics                     metrics
-	followDistance              uint64 // TODO: consider reading the finalized checkpoint from consensus layer
-	connectionTimeout           time.Duration
-	reconnectionInitialInterval time.Duration
-	reconnectionMaxInterval     time.Duration
-	logBatchSize                uint64
+	logger                            *zap.Logger
+	metrics                           metrics
+	connectionTimeout                 time.Duration
+	reconnectionInitialInterval       time.Duration
+	reconnectionMaxInterval           time.Duration
+	logBatchSize                      uint64
+	followDistance                    uint64 // used for backward compatibility only
+	finalizedCheckpointActivationSlot uint64
```

Comment:
we should get this from the networkconfig

Reply:
I think passing this as an Option is ok. By default this value contains the "INF" block number